### DISCUSSION TITLE
Armour Refactor/Balance Project

### DISCUSSION THREAD

**Discussion by Edan52** [ParadiseSS13/Paradise#18521] [Projects]
## Todo list

- [x] Remove the inquisitor suit from the tendril loot, and tone down the values on the beserker/champion's suit (remove ash storm immunity) #18698
- [x] Fix the discrepancy between the mining hardsuit armour values & the explorer suit (probably just make the explorer suit equal to the hardsuit, while maybe SLIGHTLY buffing the hardsuit) #18522
- [x] Make the H.E.C.K suit a better alternative (in most ways, at least) to the ash drake armour. #18682 
- [ ] Replace the RNG knockdown based on the force of items with something better (this considers armour, that's why it's here)
- [x] Prototype for my idea (I'm working on this) #18547
- [ ] Increase complexity for explorer armour upgrades, see Denth's comment.
- [ ] Make species armour a datum.

**Comment by Inadvizable** [comment]
Having armor stacking be based on a function (or at the very least multiplication) is generally the way to go I feel, because addition will always end up with weirdness and restrictions in designing of new things - as you have already described. There are currently checks for place that make 100+ armor impenetrable for admin purposes, which is just silly when it's entirely possible for non-admins to reach that threshold. It also reduces the potency of powergaming by the way of diminishing returns... there just isn't much point eventually. Tweaking the function lets us basically set the "maximum" armor that can be reasonably reached. I like this very much.

Different types of armor penetration could be a bit confusing for players as to what's the better option since it kinda requires doing math to figure out, which might be a bit of an ask from the average player, but I guess a meta would form soon enough. 

One discrepancy worth looking at is laser and energy armor being separate values. Lasers are generally considered "energy weapons" ICly because that just makes sense, and in fact one of the most common sources of getting blasted by a laser is the "energy gun". I wonder if there is much point in those two even being separate? It just isn't obvious which weapon deals which.

**Reply by Inadvizable** [reply]
Charlie brought up in the discord the fact that a separate energy armor allows stuff like disablers and lasers be balanced separately.

Armor-piercing could do that though. So that's something to consider if we do squash the two types of armor together.

**Comment by Denthamos** [comment]
For the goliath plating upgrade - it should require golaith hide and watcher sinew to make an upgrade kit for the armor instead of just slapping on the hide directly.  This would apply to the mech plating upgrades as well, so miners have a reason to hunt both down instead of finding a goliath tendril at the start and then being at top tier armor till they find a armor suit drop or kill a drake.

**Comment by Qwertytoforty** [comment]
Mirroring comment made on pr:
I am not sure I am a fan of this formula, that makes armor under 50 stronger and over 50 weaker. I feel like addressing the sources of stacking, sec jumpsuit, armored accessories, and bone bracers, may be better than this. This means we can keep the armor values simple to read, with reading 20 melee being a 20% reduction, vs actually being ~29% reduction. Also keeps it more consistent with how the rest of the armor works in the game. Alternatively, we make stacking armor items multiplicative, vs additive.

**Reply by hal9000PR** [reply]
if we just make it multiplicative, stacking armour would be effectively worhtless unless we make the sec jumpsuit have 30% + armour. which just aint gonna happen.
it also means we can *actually* use the physiology armour datum without miners becoming damage immune.

**Reply by Edan52** [reply]
My main point for not just nerfing the things that are currently causing issues is basically what Viz said, it creates restrictions in designing new things and lacks much depth.

I'm not sure about multiplicative armour, as this would mean that stacking different amounts of armour to the same number would result in different amounts of protection, not sure how I feel about that

I understand the point about simplicity, though I'm not too sure how to go about it without increasing the complexity of the system. By the time people figure out how to calculate damage and armour, I'd say that plugging a couple numbers into an equation is not really much more difficult than the current system.

**Reply by Inadvizable** [reply]
With multiplicative armor we don't consider two pieces of 10 armor as 20 armor, but rather as two layers of 10 armor basically. Ie. Five layers of reinforced t-shirts (let's ignore the fact that you can't layer jumpsuits) don't add up to the same protection as one metal chest plate.
